### PR TITLE
Add #all method to Lita::User, matching specs

### PR DESCRIPTION
**Pull request by chriswoodrich**

### CODE REVIEW

**Comment by jimmycuadra** [comment]
What's the use case for this?

**Comment by chriswoodrich** [comment]
Convenience for when a handler needs access to all the users for messaging
or other functionality.

File: lib/lita/user.rb [hunk 1]
```diff
@@ -69,6 +69,14 @@ def fuzzy_find(identifier)
         find_by_id(identifier) || find_by_mention_name(identifier) ||
           find_by_name(identifier) || find_by_partial_name(identifier)
       end
+
+      # Finds and returns all users for user elsewhere
+      # Returns an array of users
+      # Since 4.2.3
+      def all
+        ids = redis.keys.select { |key| key.include? "id" }.map { |id| id.split(":").last }.sort
+        ids.each.map { |id| find_by_id(id) }
```
Comment (webdestroya) [inline lib/lita/user.rb]: Should be `ids.map { |id| find_by_id(id) }`

`.each.map` is incorrect

Comment (chriswoodrich) [inline lib/lita/user.rb]: Thanks @webdestroya

**Comment by jimmycuadra** [comment]
I think we discussed this before, but unless there's something I'm not understanding about the use case, authorization groups are probably a better fit sending out mass messages. That way you can scope it to the purpose of the specific message and people it's relevant for.

It's also worth noting that there will eventually be support for a "roster" feature that will tell you who is currently online, filterable by room.

**Comment by jimmycuadra** [comment]
Thanks for this, but I think I'm going to close it in favor of a proper roster feature, which should be coming relatively soon.

**Pull request closed without merging**